crates/bevy_gltf/src/loader.rs

```diff
@@ -467,10 +469,16 @@ async fn load_gltf<'a, 'b>(
         }
 
         #[cfg(feature = "bevy_animation")]
-        if !animations.is_empty() {
-            world
-                .entity_mut(*node_index_to_entity_map.get(&0).unwrap())
-                .insert(AnimationPlayer::default());
+        {
+            // for each node root in a scene, check if it's the root of an animation
+            // if it is, add the AnimationPlayer component
+            for node in scene.nodes() {
+                if animation_roots.iter().any(|name| name == &node_name(&node)) {
```

Comment:
This feels like a _very_ heavy function, especially when scenes might have thousands of nodes:

For every node in the scene:
  * Iterate over every animation in the scene
  * Allocate a new string for the node to get its name
  * Do a string comparison between that new name and the animation root's name

Given that animations define target node indices, can we use those instead of names? Maybe a `let root_animation_nodes: HashSet<usize>`?

Reply:
`scene.nodes()` doesn't iterate over all nodes of the scene, only the root nodes. So most of the time it's only one node
https://docs.rs/gltf/1.0.0/gltf/struct.Scene.html#method.nodes

Reply:
it's not easy to get the root node indice. The animation channel is defined on the node it animates, with no easy way to get back to its root ancestor. the node tree can be traversed downward, not upward

Reply:
done 👍

Reply:
Awesome. Thanks!